library/alloc/src/rc.rs

```diff
@@ -35,15 +35,26 @@
 //! `Rc<T>` automatically dereferences to `T` (via the [`Deref`] trait),
 //! so you can call `T`'s methods on a value of type [`Rc<T>`][`Rc`]. To avoid name
 //! clashes with `T`'s methods, the methods of [`Rc<T>`][`Rc`] itself are associated
-//! functions, called using function-like syntax:
+//! functions, called using [fully qualified syntax]:
 //!
 //! ```
 //! use std::rc::Rc;
-//! let my_rc = Rc::new(());
 //!
+//! let my_rc = Rc::new(());
 //! Rc::downgrade(&my_rc);
 //! ```
 //!
+//! `Rc<T>`'s implementations of traits like `Clone` should also be called using
+//! fully qualified syntax to avoid confusion as to whether the *reference* is being
+//! cloned or the *backing data* (`T`) is being cloned:
+//!
+//! ```
+//! use std::rc::Rc;
+//!
+//! let my_rc = Rc::new(());
+//! let your_rc = Rc::clone(&my_rc);
+//! ```
```

Comment:
Note that we explicitly decided against this in https://github.com/rust-lang/rust/pull/63252.

My personal opinion is that people do use `Arc::clone` syntax in the wild, so this *should* be in the official docs one way or another, but probably without "should" or "idiomatic" langauge.

Reply:
Urgh, but #63252 didn't update

https://github.com/rust-lang/rust/blame/master/library/alloc/src/rc.rs#L64-L66

I guess, the status quo is confused here, and we should probably try to be consistent at least withing stdlib itself :)

Reply:
We teach the `::clone` version in the book too; I wasn't notified of that thread either :)

Reply:
I learned the `::clone` version (I guess from the book?) and I think it has clearer intent. Maybe for now we’ll go with it here and then we can change it later if we end up updating places like in the book?

Reply:
+1 for not using `::clone` as per https://github.com/rust-lang/rust/pull/63252
It's no longer the official guidance, and I don't think the existence of its use in other resources is good justification for continuing to use it in example code.

Reply:
still waiting for this change, @camelid . (I don't like it but it appears to be the consensus.)